### ISSUE TITLE
feat: git diff integration - map changed files to affected components

### ISSUE THREAD

**Issue by mikeshogin** [mshogin/archlint#140]
## Context

Part of graph-based test impact analysis (#137). Given a set of changed files (from `git diff`), determine which components are affected. This is the bridge between VCS and the reachability index.

## Problem

`git diff` gives file paths and changed lines. archlint works with component IDs (`package.Type.Method`). Need to map: `changed file -> component IDs in that file`.

### Step 1: Parse git diff

```bash
git diff --name-only HEAD~1 HEAD
# or
git diff --name-only main...HEAD
```

Output: list of changed file paths.

### Step 2: Map files to components

For each changed file, find all components declared in that file using the structural graph (already built by `archlint collect`):

```
internal/analyzer/go.go  ->  [analyzer.GoAnalyzer, analyzer.GoAnalyzer.Analyze, ...]
pkg/callgraph/builder.go ->  [callgraph.Builder, callgraph.Builder.Build, ...]
```

### Step 3: Optional line-level narrowing (future)

Parse `git diff --unified=0` to get changed line ranges. Filter to components whose declaration spans the changed lines. This reduces false positives significantly.

```go
type ChangedComponents struct {
    // Changed file -> components in that file
    FileToComponents map[string][]string
    // All affected component IDs (flat set)
    AffectedSet ComponentSet
}

func ComputeChanges(gitRef string, structGraph *model.Graph) (*ChangedComponents, error)
```

## Special cases

- `go.mod` or `go.sum` changed -> mark ALL components affected (conservative)
- New file added -> detect its components, add to affected set
- File deleted -> find components that imported it, add to affected set
- Renamed file -> treat as delete + add
- Test files changed -> those tests are always included regardless of reachability

## Command

```bash
# Show affected components from last commit
archlint git-diff HEAD~1

# Show affected components between branches
archlint git-diff main...HEAD

# Output machine-readable
archlint git-diff --format json HEAD~1
```

## Output

```
Changed files: 3
Affected components (package-level): 2
  - internal/analyzer
  - pkg/callgraph

go.mod unchanged: no full-rebuild required
```

## Acceptance criteria

- [ ] Parses `git diff --name-only` output correctly
- [ ] Maps file paths to component IDs using structural graph
- [ ] Handles `go.mod` change as conservative full-rebuild trigger
- [ ] Handles new/deleted/renamed files
- [ ] Returns `ComponentSet` compatible with `ReachableIndex` (#139)
- [ ] Unit tests with fixture diffs
- [ ] Works on archlint repo itself (dogfooding)

## References

- Parent issue: #137
- Depends on: #138, #139
- Art: Symflower runs `git diff` then checks package-level import graph
- Art: Bazel-diff computes affected targets between two git revisions using build graph
- Art: Microsoft TIA uses coverage data + changed lines for method-level precision
